### PR TITLE
Add filters to budget mode

### PR DESCRIPTION
**Pull request by chelseaSchmidt**
Advanced mode became filtering when I started looking more at it! What do you think of this take on it? I like it too because you can not only ignore the things you don't care about, BUT if you do care about all these things, you could choose to view the info in stages rather than all at once.

| | |
|-|-|
| image | image |
| image | image |

### CODE REVIEW

File: frontend/src/data.ts [hunk 1]
```diff
@@ -25,6 +25,13 @@ export enum Mode {
     BUDGET,
 }
 
+export enum BudgetFilter {
+    RESOURCES = "Resources",
+    PRODUCTION_LIMIT = "Production limit",
```
Comment (sirrus233) [inline frontend/src/data.ts]: ⛏️ Capitalize second words

File: frontend/src/ProductionHelper.tsx [hunk 2]
```diff
@@ -2,17 +2,28 @@ import * as React from "react";
...
 import { CalculatorModeDisplay } from "./CalculatorModeDisplay";
+import BudgetFilters from "./BudgetFilters";
 
 export default function ProductionHelper() {
     const [unitCounts, setUnitCounts] = useState<UnitCounts>(
         new Map(Object.values(Unit).map((unit) => [unit, 0]))
     );
     const [mode, setMode] = useState(Mode.CALCULATOR);
+    const [budgetFilters, setBudgetFilters] = useState(
+        new Set([BudgetFilter.RESOURCES])
```
Comment (sirrus233) [inline frontend/src/ProductionHelper.tsx]: 💭 Is this the best default? I think I might suggest resources *and* production capacity as the initial state.

File: frontend/src/ProductionHelper.tsx [hunk 3]
```diff
@@ -37,6 +48,18 @@ export default function ProductionHelper() {
         setUnitCounts(newState);
     }
 
+    function handleBudgetFiltersChange(
```
Comment (sirrus233) [inline frontend/src/ProductionHelper.tsx]: :smiling_face_with_tear:

**Review by sirrus233 — APPROVED** [review]
This is SO good, what a fantastic idea. Very impressed! I had some thoughts in this direction earlier today, and this covers everything I was thinking (and then some) but in a much nicer way. 🚀!!!

**Comment by chelseaSchmidt** [comment]
> This is SO good, what a fantastic idea. Very impressed! I had some thoughts in this direction earlier today, and this covers everything I was thinking (and then some) but in a much nicer way. 🚀!!!

@sirrus233 Awesome!! Great minds haha. Thank you!

File: frontend/src/ProductionHelper.tsx (hunk 3 above)
Comment (chelseaSchmidt) [reply frontend/src/ProductionHelper.tsx]: Crap lol, took out the `s`!

File: frontend/src/ProductionHelper.tsx (hunk 2 above)
Comment (chelseaSchmidt) [reply frontend/src/ProductionHelper.tsx]: Updated 💯

File: frontend/src/data.ts (hunk 1 above)
Comment (chelseaSchmidt) [reply frontend/src/data.ts]: Done!

**Pull request merged**